_job-scheduler-plugin/build-plugin.md

```diff
@@ -0,0 +1,122 @@
+## Parameters configuration
+
+You can configure your plugin's parameters by referencing and modifying the [example](https://github.com/opensearch-project/job-scheduler/blob/main/sample-extension-plugin/src/main/java/org/opensearch/jobscheduler/sampleextension/SampleJobParameter.java) `SampleJobParameter.java` file to fit your needs:
+
+```java
+/**
+ * A sample job parameter.
+ * <p>
+ * It adds an additional "indexToWatch" field to {@link ScheduledJobParameter}, which stores the index
+ * the job runner will watch.
+ */
+public class SampleJobParameter implements ScheduledJobParameter {
+    public static final String NAME_FIELD = "name";
+    public static final String ENABLED_FILED = "enabled";
+    public static final String LAST_UPDATE_TIME_FIELD = "last_update_time";
+    public static final String LAST_UPDATE_TIME_FIELD_READABLE = "last_update_time_field";
+    public static final String SCHEDULE_FIELD = "schedule";
+    public static final String ENABLED_TIME_FILED = "enabled_time";
+    public static final String ENABLED_TIME_FILED_READABLE = "enabled_time_field";
+    public static final String INDEX_NAME_FIELD = "index_name_to_watch";
+    public static final String LOCK_DURATION_SECONDS = "lock_duration_seconds";
+    public static final String JITTER = "jitter";
+
+    private String jobName;
+    private Instant lastUpdateTime;
+    private Instant enabledTime;
+    private boolean isEnabled;
+    private Schedule schedule;
+    private String indexToWatch;
+    private Long lockDurationSeconds;
+    private Double jitter;
+```
+
+Next, build out your parameters from the declared variables:
+
+```java
+public SampleJobParameter(String id, String name, String indexToWatch, Schedule schedule, Long lockDurationSeconds, Double jitter) {
+        this.jobName = name;
+        this.indexToWatch = indexToWatch;
+        this.schedule = schedule;
+
+        Instant now = Instant.now();
+        this.isEnabled = true;
+        this.enabledTime = now;
+        this.lastUpdateTime = now;
+        this.lockDurationSeconds = lockDurationSeconds;
+        this.jitter = jitter;
+    }
+
+    @Override
+    public String getName() {
+        return this.jobName;
+    }
+
+    @Override
+    public Instant getLastUpdateTime() {
+        return this.lastUpdateTime;
+    }
+
+    @Override
+    public Instant getEnabledTime() {
+        return this.enabledTime;
+    }
+
+    @Override
+    public Schedule getSchedule() {
+        return this.schedule;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return this.isEnabled;
+    }
+
+    @Override
+    public Long getLockDurationSeconds() {
+        return this.lockDurationSeconds;
+    }
+
+    @Override public Double getJitter() {
+        return jitter;
+    }
+```
+
+The example code above defines request parameters set when building your plugin. Reference the following table for request parameters configured by the example code above. All request parameters listed are required.
```

Comment:
You might want to move the first statement up above the examples. "The following sample shows the request parameters to set while building your plugin." The second sentence could say, "The following table describes the request parameters configured in the previous sample. All the request parameters shown are required."